src/content/ja/2019/cdn.md

```diff
@@ -0,0 +1,1567 @@
+
+最も重要なことは、これらの結果は潜在的な使用率を反映しているが、実際の影響を反映していないことです。 YouTubeは「ShoesByColin」よりも人気がありますが、使用率を比較するとどちらも同じ値として表示されます。
+
+これを念頭に置いて、CDNのコンテキストで測定されなかったいくつかの意図的な統計があります。
+
+* **TTFB**：CDNによる最初のバイトまでの時間について、キャッシュ可能とキャッシュの有効性についての正しい知識がなければ正しく測定できません。1つのサイトには、**往復時間**（RTT）の管理にCDNを使用してるがキャッシュには使用してない場合、別のCDNベンダーを使用してコンテンツをキャッシュしてる別サイトと比較する時不利になります（注：これは、個々のCDNの[パフォーマンス](./performance#time-to-first-byte-ttfb)についての結論を出していないので、パフォーマンスの章でTTFB分析には適用されません）。
+* **キャッシュヒットとキャッシュミスのパフォーマンス**：前述のようにこれは正確にテストできません、なのでコールドキャッシュとホットキャッシュでページのパフォーマンステストを繰り返すことは信頼できません。
+
+### さらなる統計
+
+Web年鑑の将来のバージョンでは、CDNベンダー間のTLSおよびRTTの管理をより詳細に検討する予定です。興味深いのは、OCSP Staplingの影響、TLS暗号パフォーマンスの違いです。 CWND（TCP輻輳ウィンドウ）成長率、特にBBR v1、v2、従来のTCP Cubicの採用。
+
+## CDNの採用と使い方
+
+ウェブサイトの場合、CDNはプライマリドメイン（`www.shoesbycolin.com`）、サブドメインまたは兄弟ドメイン（`images.shoesbycolin.com`または`checkout.shoesbycolin.com`）、そして最後にサードパーティ（Google Analyticsなど）のパフォーマンスを改善できます。これらの各ユースケースにCDNを使用すると、さまざまな方法でパフォーマンスが向上します。
+
+歴史的に、CDNは[CSS](./css)、[JavaScript](./javascript)、[画像](./media)などの静的リソース専用に使用されていました。これらのリソースはおそらくバージョン管理され（パスに一意の番号を含む）、長期的にキャッシュされます。このようにして、ベースHTMLドメインと比較して、サブドメインまたは兄弟ドメインでのCDNの採用が増加することを期待する必要があります。従来のデザインパターンでは、`www.shoesbycolin.com`がデータセンター（又は**origin**）から直接HTMLを提供し、`static.shoesbycolin.com`がCDNを使用することを想定していました。
+
+<figure>
+  <a href="/static/images/2019/17_CDN/fig1.png">
+    <img src="/static/images/2019/17_CDN/fig1.png" alt="図 1. CDN使用量 vs. originがホストするリソース" aria-labelledby="fig1-caption" aria-describedby="fig1-description" width="600" data-width="600" data-height="371" data-seamless data-frameborder="0" data-scrolling="no" data-src="https://docs.google.com/spreadsheets/d/e/2PACX-1vRzPn-1SGVa3rNCT0U9QeQNODE97fsmXyaJX1ZOoBNR8nPpclhC6fg8R_UpoodeiX6HkdHrp50WBQ5Q/pubchart?oid=777938536&format=interactive">
+  </a>
+  <div id="fig1-description" class="visually-hidden">HTMLがoriginから提供される80％、CDNから20％、サブドメインが61％と39％、サードパーティが34％と66％である積層棒グラフ。</div>
+  <figcaption id="fig1-caption">図 1. CDN使用量 vs. originがホストするリソース</figcaption>
+</figure>
+
+実際、この伝統的なパターンは、クロールされたWebサイトの大部分で見られるものです。 Webページの大部分（80％）は、元のベースHTMLを提供しています。この内訳はモバイルとデスクトップでほぼ同じであり、デスクトップでのCDNの使用率は0.4％しか低下していません。このわずかな差異は、CDNをより頻繁に使用するモバイル固有のWebページ（「mDot」）の小規模な継続使用に起因する可能性があります。
+
+同様に、サブドメインから提供されるリソースは、サブドメインリソースの40％でCDNを利用する可能性が高くなります。サブドメインは、画像やCSSなどのリソースを分割するために使用されるか、チェックアウトやAPIなどの組織チームを反映するために使用されます。
+
+ファーストパーティのリソースの大部分は依然としてoriginから直接提供されていますが、サードパーティのリソースはCDNの採用が大幅に増えています。すべてのサードパーティリソースのほぼ66％がCDNから提供されています。サードパーティのドメインはSaaS統合である可能性が高いため、CDNの使用はこれらのビジネス製品のコアになる可能性が高くなります。ほとんどのサードパーティコンテンツは共有リソース（JavaScriptまたはフォントCDN）、拡張コンテンツ（広告）、または統計に分類されます。これらすべての場合においてCDNを使用すると、SaaSソリューションのパフォーマンスとオフロードが向上します。
+
+## トップCDNプロバイダー
+
```

Comment:
Extra blank line that doesn't exist in English version?

```suggestion
```

Reply:
Thanks, I fix.